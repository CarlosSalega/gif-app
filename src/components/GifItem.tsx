import { ShowImage } from "../helpers/types";

const GifItem = ({ title, images }: ShowImage) => {
  return (
    <div className="card">
      <img src={images.downsized_medium.url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GifItem;
