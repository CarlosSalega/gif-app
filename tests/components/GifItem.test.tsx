import React from "react";
import { render } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Pruebas en <GifItem />", () => {
  const title = "Example Title";
  const url = "https://example.com/image.gif";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(
      <GifItem title={title} images={{ downsized_medium: { url } }} />
    );
    expect(container).toMatchSnapshot();
  });
});
