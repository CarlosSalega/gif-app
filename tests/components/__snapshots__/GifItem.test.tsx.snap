// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Example Title"
      src="https://example.com/image.gif"
    />
    <p>
      Example Title
    </p>
  </div>
</div>
`;
